### PR TITLE
Incorporate Detection and Identification Rate and Curve

### PR DESCRIPTION
**Pull request by siebenkopf**
After some discussion:
https://groups.google.com/forum/?fromgroups#!topic/bob-devel/TL6ODJ_WPTA
we have come up with a solution for measures for open set recognition, which I have implemented. Time to merge...

### CODE REVIEW

**Comment by siebenkopf** [comment]
I'll give you some more time to comment on this, and I am going to merge it on Monday.

File: bob/measure/__init__.py [hunk 1]
```diff
@@ -79,62 +79,108 @@ def relevance (input, machine):
   return retval
 
 
-def recognition_rate(cmc_scores, threshold=None):
-  """recognition_rate(cmc_scores, threshold) -> RR 
-  
+def recognition_rate(cmc_scores, threshold = None, rank = 1):
```
Comment (183amir) [inline bob/measure/__init__.py]: You **should not** add spaces around the `=` operator here according to `PEP8`. Please revert this back and remove extra spaces in your optional variables of your new function definitions.
My suggestion is that you use a linter to fix these issues. I usually use pep8 autoformat to fix my code while ignoring these rules:

    "autoformat_ignore":
    [
        "E309",
        "E111",
        "E114",
        "E501",
        "E121",
        "E101",
    ],

I don't know the exact command to run pep8 autoformat I use this IDE: https://damnwidget.github.io/anaconda/ and run it through that.

**Comment by tiagofrepereira2012** [comment]
Hi @siebenkopf , thanks for the pull.
I've checked already the code, for me you can merge it.

Hey @183amir , let's keep this recomendation for future commits.
I think it's a bit too much to revert pull requests because of white spaces.

Furthermore, if you go crazy with this PEP, we will need to revert all bob.
For instance, they recommend to use 4 spaces for identation and we use 2.

**Comment by 183amir** [comment]
Hey @tiagofrepereira2012, by reverting it I meant remove those extra whitespace not reverting the whole pull request :D

**Comment by 183amir** [comment]
And as for using 4 spaces instead of 2 and other things we do, that's why I also added an ignore list which for examples ignores 2 space situation.

**Comment by 183amir** [comment]
Also it's not the lint that bothers me, in some of our code I have seen a lot of unused imports, unused variables and that leads to bugs sometimes and slowdown in code. If you were using some kind of IDE this would be prevented. Something like this could come in handy: https://pypi.python.org/pypi/flake8

**Comment by siebenkopf** [comment]
Honestly, I don't care too much about PEP and their indentation and spacing rules. 
As long as it is not **enforced** by the language, I think we should be free to do, whatever we want. And I think, e.g., 2 spaces indentation and having spaces around assignment operators (independent on where in the code) makes the code more readable.
This also includes that I am generally not breaking my code at 80 characters. Modern editors can do that for you, without adding actual brakes in the code. If your editor cannot do that, you should think about switching :-)

**Comment by anjos** [comment]
If we'd like code to respect certain criteria, I'd suggest we create a lint test for it (using pyflakes, pep8, pylint or whatever). Forcing people to use specific editors is not a good approach. It is, unfortunately, taken like a religion - everyone thinks they're right and no other solution exists.

**Pull request merged**